src/logcollector/read_win_event_channel.c

```diff
@@ -522,12 +523,22 @@ DWORD WINAPI event_channel_callback(EVT_SUBSCRIBE_NOTIFY_ACTION action, os_chann
 {
     if (action == EvtSubscribeActionDeliver) {
         send_channel_event(evt, channel);
+    } else {
+        while(1) {
+            /* Try to restart EventChannel */
+            if (win_start_event_channel(channel->evt_log, !channel->bookmark_enabled, channel->query) == -1) {
+                sleep(5);
```

Comment:
Let's make this sleep configurable, include an option readable in the ossec.conf called `reconnect_time` for each configured channel.